assets/src/scripts/components/graph/view/index.js

```diff
@@ -108,14 +112,27 @@ const drawChart = function (elem, store, opts, chartType) {
                     yScale: getScaleY(opts, chartType)
                 }))));
         })
-        // Draw the y-axis, only for the main chart.
+        // Draw the y-axis for the main.
         .call(callIf(chartType === 'main', link(store, drawAxisY, createStructuredSelector({
             yScale: getScaleY(opts, chartType),
             layout: getChartPosition(opts, chartType)
         }), (bBox) => {
             // When the bounding box has changed, update the state with it.
             store.dispatch(setAxisYBBox(opts.id, bBox));
         })))
+        // Draw the y-axis for well depth on the lithology chart (which is part of the well diagram)
+        .call(callIf(chartType === 'lithology', link(store, drawAxisYWellDiagramDepth, createStructuredSelector({
+            yScale: getScaleY(opts, chartType),
+            layout: getChartPosition(opts, chartType)
+        }), (bBox) => {
```

Comment:
This worries me. Do you understand why this is necessary? Seems like a state change will eventually generate another state change and this seems wrong. Maybe we can talk about this if you are in the office tomorrow.

Reply:
This is used to crop the SVG viewbox to the actual size of the chart, so you don't have to guess how wide the axis is. You don't know how wide it will be until after it has been drawn.

I'd suggest using this method with the right axis also, or trying to come up with a way to get the bounding box of the entire SVG at once. The challenge with the latter is determining events to listen to that won't recalculate the bounding box excessively, when nothing has changed.